### DISCUSSION TITLE
Support of OpenAPI 3.1.0

### DISCUSSION THREAD

**Discussion by fidgi** [schemathesis/schemathesis#1397] [Q&A] [answered]
Hi,

Is there any plan to support the OpenAPI version 3.1.0 ?

regards,

**Comment by Stranger6667** [comment, marked as answer]
Yep, we plan to add support for it! Tracking issue: #494

At the moment, Schemathesis provides an option to force schema version which makes it possible to pass 3.1.0 as 3.0.3. Likely, in many aspects, it will work and will ignore new keywords from JSON Schema. Though some keywords that changed their behavior (like items) won't work properly + I am not sure about other 3.1.0-specific things. However, I don't expect many incompatibilities on the Open API level itself, mostly JSON Schema support is the issue.
So, it might work this way, but there are no guarantees 😦

We mostly depend on other libraries that need to support newer JSON Schema drafts first. There aren't that many things in `jsonschema` to finish and they are quite niche (dynamicRef / recursiveRef), but then `hypothesis-jsonschema` doesn't support any new keywords yet, and we wanted to implement some refactorings first, so it is quite complicated there.
